web/src/pages/api/user/login.ts

```diff
@@ -2,53 +2,57 @@ import { compare } from "bcrypt";
 import { NextApiRequest, NextApiResponse } from "next";
 
 import { prisma } from "../../../server/db/client";
+import sendConfirmationEmail from "../../../utils/email";
 import { generateJWT } from "../../../utils/jwt";
 import { validateMethod } from "../../../utils/validation";
 
 interface Request extends NextApiRequest {
   body: {
-    email: string;
+    email?: string;
+    name?: string;
     password: string;
   };
 }
 
 export default async function Login(req: Request, res: NextApiResponse) {
   if (!validateMethod("POST", req, res)) return;
 
-  const { email, password } = req.body;
+  const { email, name, password } = req.body;
 
-  if (!email.trim() || !password.trim()) {
+  if (!(email?.trim() || name?.trim()) || !password.trim()) {
```

Comment:
This reads better as `if (!email?.trim() || !name?.trim() || !password.trim())`

Reply:
I believe that wouldn't work in this case. I'm allowing the consumer to send either a name OR email. So if they send an email without a name it should proceed as both are unique and we can find the user with either. However, they should always send a password

Reply:
we can apply de morgan's and get `!email?.trim() && !name?.trim() || !password.trim()` but it is arguable which is more readable

Reply:
Ah I see. I can't do boolean logic.